packages/mapper/src/ProvidesCaster.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Tempest\Mapper;
+
+/**
+ * Implemented on an attribute, this interface specifies that a specific caster must be used to serialize the associated property.
+ */
+interface ProvidesCaster
+{
+    /** @var class-string<Caster> */
+    public string $caster {
```

Comment:
Not a dealbreaker but I tend to suffix variables with `class` when they represent a class string. So `$casterClass` and `$serializerClass`.